build-data-capturing-gradle-samples/capture-quality-check-issues/gradle-quality-check-issues.gradle

```diff
@@ -10,100 +13,143 @@ if (!buildScanApi) {
-        valueName = 'Verification Checkstyle'
-        errors = report.file.collect {
-            String filePath = task.project.rootProject.relativePath(it.@name.text())
-            it.error.collect { "${filePath}:${it.@line}:${it.@column} \u2192 ${it.@message}" }
-        }.flatten() as List<String>
-    } else if (ReportingTask.CODENARC.isTask(task)) {
-        valueName = 'Verification CodeNarc'
-        errors = report.Package.collect {
-            def sourceDirectory = appendIfMissing(it.parent().Project.SourceDirectory.text() as String, '/')
-            it.File.collect {
-                String filePath = task.project.rootProject.relativePath(sourceDirectory + (it.@name.text() as String))
-                it.Violation.collect { "${filePath}:${it.@lineNumber} \u2192 ${it.Message.text()}" }
-            }.flatten()
-        }.flatten() as List<String>
-    } else if (ReportingTask.FINDBUGS.isTask(task)) {
-        valueName = 'Verification FindBugs'
-        errors = report.BugInstance.collect { bugInstance ->
-            String type = bugInstance.@type.text()
-            def error = bugInstance.breadthFirst().findAll { it.name() == 'SourceLine' }.sort(false) { it.parent().name() as SpotBugsParent }.first()
-            String startLine = error.@start.text()
-            String endLine = error.@end.text()
-            String lineNumber = endLine == startLine ? startLine : "$startLine-$endLine"
-            def absoluteFilePath = bugInstance.parent().Project.SrcDir.collect { it.text() }.find { it.endsWith(error.@sourcepath.text()) }
-            String relativePath = task.project.rootProject.relativePath(absoluteFilePath)
-            "${relativePath}:${lineNumber} \u2192 ${type}"
-        }.flatten() as List<String>
-    } else if (ReportingTask.SPOTBUGS.isTask(task)) {
-        valueName = 'Verification SpotBugs'
-        errors = report.BugInstance.collect { bugInstance ->
-            String type = bugInstance.@type.text()
-            def error = bugInstance.breadthFirst().findAll { it.name() == 'SourceLine' }.sort(false) { it.parent().name() as SpotBugsParent }.first()
-            String startLine = error.@start.text()
-            String endLine = error.@end.text()
-            String lineNumber = endLine == startLine ? startLine : "$startLine-$endLine"
-            def absoluteFilePath = bugInstance.parent().Project.SrcDir.collect { it.text() }.find { it.endsWith(error.@sourcepath.text()) }
-            String relativePath = task.project.rootProject.relativePath(absoluteFilePath)
-            "${relativePath}:${lineNumber} \u2192 ${type}"
-        }.flatten() as List<String>
-    }
-
-    errors.each { buildScan.value valueName, it }
 }
 
-static String appendIfMissing(String str, String suffix) {
-    str.endsWith(suffix) ? str : str + suffix
+// check if reporting tasks are present in the build
+def hasReportingTask = false
+gradle.taskGraph.whenReady { taskGraph ->
+    if(taskGraph.allTasks.findAll {
```

Comment:
does this `findAll` eagerly configure tasks?

Reply:
I don't think so, but this will vanish as well anyway